json.go

```diff
@@ -0,0 +1,174 @@
+package feeds
+
+import (
+	"encoding/json"
+	"time"
+)
+
+const jsonFeedVersion = "https://jsonfeed.org/version/1"
+
+// JSONAuthor represents the author of the feed or of an individual item
+// in the feed
+type JSONAuthor struct {
+	Name   string `json:"name,omitempty"`
+	Url    string `json:"url,omitempty"`
+	Avatar string `json:"avatar,omitempty"`
+}
+
+// JSONAttachment represents a related resource. Podcasts, for instance, would
+// include an attachment that’s an audio or video file.
+type JSONAttachment struct {
+	Url      string        `json:"url,omitempty"`
+	MIMEType string        `json:"mime_type,omitempty"`
+	Title    string        `json:"title,omitempty"`
+	Size     int32         `json:"size,omitempty"`
+	Duration time.Duration `json:"duration_in_seconds,omitempty"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+// The Duration field is marshaled in seconds, all other fields are marshaled
+// based upon the definitions in struct tags.
+func (a *JSONAttachment) MarshalJSON() ([]byte, error) {
+	type EmbeddedJSONAttachment JSONAttachment
+	return json.Marshal(&struct {
+		Duration float64 `json:"duration_in_seconds,omitempty"`
+		*EmbeddedJSONAttachment
+	}{
+		EmbeddedJSONAttachment: (*EmbeddedJSONAttachment)(a),
+		Duration:               a.Duration.Seconds(),
+	})
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+// The Duration field is expected to be in seconds, all other field types
+// match the struct definition.
+func (a *JSONAttachment) UnmarshalJSON(data []byte) error {
+	type EmbeddedJSONAttachment JSONAttachment
+	var raw struct {
+		Duration float64 `json:"duration_in_seconds,omitempty"`
+		*EmbeddedJSONAttachment
+	}
+	raw.EmbeddedJSONAttachment = (*EmbeddedJSONAttachment)(a)
+
+	err := json.Unmarshal(data, &raw)
+	if err != nil {
+		return err
+	}
+
+	if raw.Duration > 0 {
+		nsec := int64(raw.Duration * float64(time.Second))
+		raw.EmbeddedJSONAttachment.Duration = time.Duration(nsec)
+	}
+
+	return nil
+}
+
+// JSONItem represents a single entry/post for the feed.
+type JSONItem struct {
+	Id            string           `json:"id"`
+	Url           string           `json:"url,omitempty"`
```

Comment:
Not changed based on comment @kisielk about remaining internally consistent with other names in this package.